diff -uNr qemu-orig/block.c qemu/block.c
--- qemu-orig/block.c	2023-03-01 23:51:40.543615983 -0300
+++ qemu/block.c	2023-03-02 00:36:53.073612694 -0300
@@ -888,8 +888,8 @@
      * so the files can become very large. /tmp is often a tmpfs where as
      * /var/tmp is usually on a disk, so more appropriate for disk images.
      */
-    if (!g_strcmp0(tmpdir, "/tmp")) {
-        tmpdir = "/var/tmp";
+    if (!g_strcmp0(tmpdir, "@TERMUX_PREFIX@/tmp")) {
+        tmpdir = "@TERMUX_PREFIX@/tmp";
     }
 #endif
 
diff -uNr qemu-orig/hw/hppa/machine.c qemu/hw/hppa/machine.c
--- qemu-orig/hw/hppa/machine.c	2023-03-01 23:51:38.543615954 -0300
+++ qemu/hw/hppa/machine.c	2023-03-02 00:10:33.893613722 -0300
@@ -130,19 +130,19 @@
     fw_cfg_add_i64(fw_cfg, FW_CFG_RAM_SIZE, ms->ram_size);
 
     val = cpu_to_le64(MIN_SEABIOS_HPPA_VERSION);
-    fw_cfg_add_file(fw_cfg, "/etc/firmware-min-version",
+    fw_cfg_add_file(fw_cfg, "@TERMUX_PREFIX@/etc/firmware-min-version",
                     g_memdup(&val, sizeof(val)), sizeof(val));
 
     val = cpu_to_le64(HPPA_TLB_ENTRIES);
-    fw_cfg_add_file(fw_cfg, "/etc/cpu/tlb_entries",
+    fw_cfg_add_file(fw_cfg, "@TERMUX_PREFIX@/etc/cpu/tlb_entries",
                     g_memdup(&val, sizeof(val)), sizeof(val));
 
     val = cpu_to_le64(HPPA_BTLB_ENTRIES);
-    fw_cfg_add_file(fw_cfg, "/etc/cpu/btlb_entries",
+    fw_cfg_add_file(fw_cfg, "@TERMUX_PREFIX@/etc/cpu/btlb_entries",
                     g_memdup(&val, sizeof(val)), sizeof(val));
 
     val = cpu_to_le64(HPA_POWER_BUTTON);
-    fw_cfg_add_file(fw_cfg, "/etc/power-button-addr",
+    fw_cfg_add_file(fw_cfg, "@TERMUX_PREFIX@/etc/power-button-addr",
                     g_memdup(&val, sizeof(val)), sizeof(val));
 
     fw_cfg_add_i16(fw_cfg, FW_CFG_BOOT_DEVICE, ms->boot_config.order[0]);
diff -uNr qemu-orig/hw/usb/ccid-card-emulated.c qemu/hw/usb/ccid-card-emulated.c
--- qemu-orig/hw/usb/ccid-card-emulated.c	2023-03-01 23:51:38.413615952 -0300
+++ qemu/hw/usb/ccid-card-emulated.c	2023-03-02 00:11:23.903612725 -0300
@@ -417,7 +417,7 @@
     event_notifier_cleanup(&card->notifier);
 }
 
-#define CERTIFICATES_DEFAULT_DB "/etc/pki/nssdb"
+#define CERTIFICATES_DEFAULT_DB "@TERMUX_PREFIX@/etc/pki/nssdb"
 #define CERTIFICATES_ARGS_TEMPLATE\
     "db=\"%s\" use_hw=no soft=(,Virtual Reader,CAC,,%s,%s,%s)"
 
diff -uNr qemu-orig/linux-user/flatload.c qemu/linux-user/flatload.c
--- qemu-orig/linux-user/flatload.c	2023-03-01 23:51:38.683615956 -0300
+++ qemu/linux-user/flatload.c	2023-03-02 00:34:44.123616236 -0300
@@ -692,7 +692,7 @@
 	char buf[16];
 
 	/* Create the file name */
-	sprintf(buf, "/lib/lib%d.so", id);
+	sprintf(buf, "@TERMUX_PREFIX@/lib/lib%d.so", id);
 
 	/* Open the file up */
 	bprm.filename = buf;
diff -uNr qemu-orig/linux-user/syscall.c qemu/linux-user/syscall.c
--- qemu-orig/linux-user/syscall.c	2023-03-01 23:51:38.703615956 -0300
+++ qemu/linux-user/syscall.c	2023-03-02 00:12:24.213610665 -0300
@@ -8334,7 +8334,7 @@
             /* create temporary file to map stat to */
             tmpdir = getenv("TMPDIR");
             if (!tmpdir)
-                tmpdir = "/tmp";
+                tmpdir = "@TERMUX_PREFIX@/tmp";
             snprintf(filename, sizeof(filename), "%s/qemu-open.XXXXXX", tmpdir);
             fd = mkstemp(filename);
             if (fd < 0) {
diff -uNr qemu-orig/migration/exec.c qemu/migration/exec.c
--- qemu-orig/migration/exec.c	2023-03-01 23:51:39.563615969 -0300
+++ qemu/migration/exec.c	2023-03-02 00:12:57.623610958 -0300
@@ -28,7 +28,7 @@
 void exec_start_outgoing_migration(MigrationState *s, const char *command, Error **errp)
 {
     QIOChannel *ioc;
-    const char *argv[] = { "/bin/sh", "-c", command, NULL };
+    const char *argv[] = { "@TERMUX_PREFIX@/bin/sh", "-c", command, NULL };
 
     trace_migration_exec_outgoing(command);
     ioc = QIO_CHANNEL(qio_channel_command_new_spawn(argv,
@@ -55,7 +55,7 @@
 void exec_start_incoming_migration(const char *command, Error **errp)
 {
     QIOChannel *ioc;
-    const char *argv[] = { "/bin/sh", "-c", command, NULL };
+    const char *argv[] = { "@TERMUX_PREFIX@/bin/sh", "-c", command, NULL };
 
     trace_migration_exec_incoming(command);
     ioc = QIO_CHANNEL(qio_channel_command_new_spawn(argv,
diff -uNr qemu-orig/net/tap.c qemu/net/tap.c
--- qemu-orig/net/tap.c	2023-03-01 23:51:38.743615957 -0300
+++ qemu/net/tap.c	2023-03-02 00:13:33.003614342 -0300
@@ -551,7 +551,7 @@
             *parg++ = helper_cmd;
             *parg++ = NULL;
 
-            execv("/bin/sh", args);
+            execv("@TERMUX_PREFIX@/bin/sh", args);
             g_free(helper_cmd);
         } else {
             /* assume helper is just the executable path name */
diff -uNr qemu-orig/qemu-nbd.c qemu/qemu-nbd.c
--- qemu-orig/qemu-nbd.c	2023-03-01 23:51:38.343615951 -0300
+++ qemu/qemu-nbd.c	2023-03-02 00:15:01.233618806 -0300
@@ -57,7 +57,6 @@
 #define HAVE_NBD_DEVICE 0
 #endif
 
-#define SOCKET_PATH                "/var/lock/qemu-nbd-%s"
 #define QEMU_NBD_OPT_CACHE         256
 #define QEMU_NBD_OPT_AIO           257
 #define QEMU_NBD_OPT_DISCARD       258
@@ -98,7 +97,7 @@
 "  -p, --port=PORT           port to listen on (default `%d')\n"
 "  -b, --bind=IFACE          interface to bind to (default `0.0.0.0')\n"
 "  -k, --socket=PATH         path to the unix socket\n"
-"                            (default '"SOCKET_PATH"')\n"
+"                            (default '@TERMUX_PREFIX@/tmp/nbd-%s')\n"
 "  -e, --shared=NUM          device can be shared by NUM clients (default '1')\n"
 "  -t, --persistent          don't exit on the last connection\n"
 "  -v, --verbose             display extra debugging information\n"
@@ -981,7 +980,7 @@
 
     if (device != NULL && sockpath == NULL) {
         sockpath = g_malloc(128);
-        snprintf(sockpath, 128, SOCKET_PATH, basename(device));
+        snprintf(sockpath, 128, "@TERMUX_PREFIX@/tmp/nbd-%s", basename(device));
     }
 
     server = qio_net_listener_new();
diff -uNr qemu-orig/util/module.c qemu/util/module.c
--- qemu-orig/util/module.c	2023-03-01 23:51:39.403615966 -0300
+++ qemu/util/module.c	2023-03-02 00:39:15.413609399 -0300
@@ -242,7 +242,7 @@
     version_dir = g_strcanon(g_strdup(QEMU_PKGVERSION),
                              G_CSET_A_2_Z G_CSET_a_2_z G_CSET_DIGITS "+-.~",
                              '_');
-    dirs[n_dirs++] = g_strdup_printf("/var/run/qemu/%s", version_dir);
+    dirs[n_dirs++] = g_strdup_printf("@TERMUX_PREFIX@/var/run/qemu/%s", version_dir);
 #endif
     assert(n_dirs <= ARRAY_SIZE(dirs));
 
