diff -uNr qemu-orig/include/qapi/util.h qemu/include/qapi/util.h
--- qemu-orig/include/qapi/util.h	2023-03-02 01:44:18.973613382 -0300
+++ qemu/include/qapi/util.h	2023-03-02 01:46:15.183614480 -0300
@@ -11,6 +11,10 @@
 #ifndef QAPI_UTIL_H
 #define QAPI_UTIL_H
 
+#include <glib.h>
+
+typedef struct Error Error;
+
 typedef enum {
     QAPI_DEPRECATED,
     QAPI_UNSTABLE,
diff -uNr qemu-orig/meson.build qemu/meson.build
--- qemu-orig/meson.build	2023-03-02 01:44:19.213613374 -0300
+++ qemu/meson.build	2023-03-02 01:45:12.483607879 -0300
@@ -207,7 +207,7 @@
 
 # Specify linker-script with add_project_link_arguments so that it is not placed
 # within a linker --start-group/--end-group pair
-if get_option('fuzzing')
+if false
   add_project_link_arguments(['-Wl,-T,',
                               (meson.current_source_dir() / 'tests/qtest/fuzz/fork_fuzz.ld')],
                              native: false, language: ['c', 'cpp', 'objc'])
@@ -3140,8 +3140,8 @@
 subdir('linux-user')
 
 # needed for fuzzing binaries
-subdir('tests/qtest/libqos')
-subdir('tests/qtest/fuzz')
+#subdir('tests/qtest/libqos')
+#subdir('tests/qtest/fuzz')
 
 # accel modules
 tcg_real_module_ss = ss.source_set()
