--- a/dlls/xtajit/unixlib.c
+++ b/dlls/xtajit/unixlib.c
@@ -71,7 +71,7 @@
     char *qemu_log;
     int log_mask;
 
-    if (!(emuapi_handle = dlopen( "/opt/libqemu-i386.so", RTLD_NOW ))) return STATUS_DLL_NOT_FOUND;
+    if (!(emuapi_handle = dlopen( "@TERMUX_PREFIX@/opt/hangover/lib/libqemu-i386.so", RTLD_NOW ))) return STATUS_DLL_NOT_FOUND;
 
 #define LOAD_FUNCPTR(f) if((p##f = dlsym(emuapi_handle, #f)) == NULL) {ERR(#f " %p\n", p##f);return STATUS_ENTRYPOINT_NOT_FOUND;}
 #define LOAD_FUNCPTR_OPT(f) if((p##f = dlsym(emuapi_handle, #f)) == NULL) {ERR(#f " %p\n", p##f);}
