--- a/dlls/xtajit/unixlib.c
+++ b/dlls/xtajit/unixlib.c
@@ -62,6 +62,7 @@
 static void (*ppage_set_flags)(target_ulong start, target_ulong end, int flags);
 static void (*pcpu_sync_bndcs_hflags)(CPUX86State *env);
 static void (*ptb_invalidate_phys_page_range)(tb_page_addr_t start, tb_page_addr_t end);
+static void** (*plibqemu_get_tcg_ctx)(void);
 
 static void* *ptcg_ctx;
 
@@ -117,1 +117,2 @@
-    LOAD_FUNCPTR(tcg_ctx);
+    LOAD_FUNCPTR(libqemu_get_tcg_ctx);
+    ptcg_ctx = plibqemu_get_tcg_ctx();
